### ISSUE TITLE
support maven.compiler.release property to control javac and scalac options

### ISSUE THREAD

**Issue by gzm55** [davidB/scala-maven-plugin#435]
When compile with JDK 11, and generate objects targeting JVM 8, we should be able to config `javac -release ` and `scala-2.13 -release` options by `maven.compiler.release` property, like the [`maven-compiler-plugin`](https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#release).

**Comment by slandelle** [comment]
Have you tried `<addJavacArgs>` as described in the [documentation](https://davidb.github.io/scala-maven-plugin/compile-mojo.html)?

**Comment by gzm55** [comment]
> Have you tried `<addJavacArgs>` as described in the [documentation](https://davidb.github.io/scala-maven-plugin/compile-mojo.html)?

yes, i have tried, but the properties are more convenient to control compiler options for jdk*scala version matrix.

**Comment by slandelle** [comment]
Sorry, I don't get it. Feel free to contribute a PR.

**Comment by gzm55** [comment]
@slandelle add an fast fix pr

**Comment by slandelle** [comment]
> yes, i have tried, but the properties are more convenient to control compiler options for jdk*scala version matrix.

Could you please elaborate on this statement? I don't get how it's that much more convenient it's worth introducing new dedicated options.

**Comment by gzm55** [comment]
We have a project built by jdk 8 or 11, and release for scala 1.12 and scala 1.13, with the compile target jvm 8 for both java and scala sources. We need carefully deal with the `javac` and `scalac` options, since

* from JDK 9, javac should be passed `--release 8` instead of `-source 8 -target 8`
* from scala 2.13, we should use `-target:8` instead of `-target:jvm-1.8` for `scalac`
* from scala 2.13 & JDK9+, `--release 8` should be passed to `scalac`

Then in the `pom.xml` with the current scala-maven-plugin, we have to:

#### profile jdk8+scala2.12:

* compiler.source = 1.8
* compiler.target = 1.8
* args = [ -target:jvm-1.8 ]

#### profile jdk11+scala2.12:

* args = [ -target:jvm-1.8 ]
* javacArgs = [ --release, 8 ]

#### profile jdk8+scala2.13:

* compiler.source = 1.8
* compiler.target = 1.8
* args = [ -target:8 ]

#### profile jdk11+scala2.13:

* args = [ -target:8, --release, 8 ]
* javacArgs = [ --release, 8 ]

That is, we have to config for each combination of scala and jdk versions, deal the configurations relation in pom, in addition,  the `args` and `jaracArgs` lists may not be reused by children poms (override the whole args list).

Suppose the scala-maven-plugin can understand specific configuration keys for the `release` option of `javac`/`scalac`, and deal their releation like maven-compiler-plugin, then the profiles can be simplified as:

#### default properties:

* compiler.source = 1.8
* compiler.target = 1.8

#### profile jdk11:

* compiler.release = 8 # note the `release` option of `javac` should override the `source` and `target`

#### profile scala-2.12

* scala.target = jvm-1.8

#### profile scala-2.13

* scala.target = 8
* scala.release = 8

**Comment by gzm55** [comment]
@slandelle any updates?
